test/boostTest.cpp

```diff
@@ -162,71 +162,89 @@ int registerTests(
+		return false;
 	options->validate();
 
 	solidity::test::CommonOptions::setSingleton(std::move(options));
+	return true;
 }
 }
 
 // TODO: Prototype -- why isn't this declared in the boost headers?
 // TODO: replace this with a (global) fixture.
-test_suite* init_unit_test_suite( int /*argc*/, char* /*argv*/[] );
+test_suite* init_unit_test_suite(int /*argc*/, char* /*argv*/[]);
 
-test_suite* init_unit_test_suite( int /*argc*/, char* /*argv*/[] )
+test_suite* init_unit_test_suite(int /*argc*/, char* /*argv*/[])
 {
 	master_test_suite_t& master = framework::master_test_suite();
 	master.p_name.value = "SolidityTests";
 
-	initializeOptions();
+	try
+	{
+		bool shouldContinue = initializeOptions();
+		if (!shouldContinue)
+			exit(0);
 
-	if (!solidity::test::loadVMs(solidity::test::CommonOptions::get()))
-		exit(1);
+		if (!solidity::test::loadVMs(solidity::test::CommonOptions::get()))
+			exit(1);
+
+		if (solidity::test::CommonOptions::get().disableSemanticTests)
+			cout << endl << "--- SKIPPING ALL SEMANTICS TESTS ---" << endl << endl;
+
+		// Include the interactive tests in the automatic tests as well
+		for (auto const& ts: g_interactiveTestsuites)
+		{
+			auto const& options = solidity::test::CommonOptions::get();
+
+			if (ts.smt && options.disableSMT)
+				continue;
 
-	if (solidity::test::CommonOptions::get().disableSemanticTests)
-		cout << endl << "--- SKIPPING ALL SEMANTICS TESTS ---" << endl << endl;
+			if (ts.needsVM && solidity::test::CommonOptions::get().disableSemanticTests)
+				continue;
 
-	// Include the interactive tests in the automatic tests as well
-	for (auto const& ts: g_interactiveTestsuites)
+			solAssert(registerTests(
+				master,
+				options.testPath / ts.path,
+				ts.subpath,
+				options.enforceViaYul,
+				options.enforceCompileToEwasm,
+				ts.labels,
+				ts.testCaseCreator
+			) > 0, std::string("no ") + ts.title + " tests found");
+		}
+
+		if (solidity::test::CommonOptions::get().disableSemanticTests)
+		{
+			for (auto suite: {
+				"ABIDecoderTest",
+				"ABIEncoderTest",
+				"SolidityAuctionRegistrar",
+				"SolidityFixedFeeRegistrar",
+				"SolidityWallet",
+				"GasMeterTests",
+				"GasCostTests",
+				"SolidityEndToEndTest",
+				"SolidityOptimizer"
+			})
+				removeTestSuite(suite);
+		}
+	}
+	catch (solidity::test::ConfigException const& exception)
 	{
-		auto const& options = solidity::test::CommonOptions::get();
-
-		if (ts.smt && options.disableSMT)
-			continue;
-
-		if (ts.needsVM && solidity::test::CommonOptions::get().disableSemanticTests)
-			continue;
-
-		solAssert(registerTests(
-			master,
-			options.testPath / ts.path,
-			ts.subpath,
-			options.enforceViaYul,
-			options.enforceCompileToEwasm,
-			ts.labels,
-			ts.testCaseCreator
-		) > 0, std::string("no ") + ts.title + " tests found");
+		cerr << exception.what() << endl;
+		exit(1);
 	}
-
-	if (solidity::test::CommonOptions::get().disableSemanticTests)
+	catch (std::runtime_error const& exception)
```

Comment:
Because the exceptions we throw from assertions inherit from `exception` and I don't want to lump them together. `runtime_error` seems to be used for reporting validation errors and if we catch one, `isoltest` should just print the message and exit. This is normal and can happen e.g. if you just use wrong arguments, file is missing, etc. But if we get an assertion or another unexpected exception that should have been handled earlier we have a bug - we want to defer to the top-level exception handler, which will just print as much debug info as it can and exit.